helix-view/src/tree.rs

```diff
@@ -253,6 +253,18 @@ impl Tree {
         })
     }
 
+    pub fn views_with_keys_mut(&mut self) -> impl Iterator<Item = (&mut View, ViewId)> {
```

Comment:
This method is unnecessary, you can already access the id via view.id

Reply:
All `Tree` fields apart from `focus` are private AFAIK:

`helix-view/src/tree.rs:7`
```rs
pub struct Tree {
    root: ViewId,
    // (container, index inside the container)
    pub focus: ViewId,
    // fullscreen: bool,
    area: Rect,

    nodes: HopSlotMap<ViewId, Node>,

    // used for traversals
    stack: Vec<(ViewId, Rect)>,
}
```

Reply:
Why do you need access to nodes?

Reply:
The body inside MouseEventKind::Down can simply be something like:

```rust
let result = tree.views_mut().find_map(|(view, _focus)| {
  // no jumps
  let doc = &mut editor.documents[view.doc];
  view.pos_at_screen_coords(doc, row, column).map(|coords| (view.id, coords))
})

if let Some((view_id, coords)) = result {
   ... set_selection etc
}
```

Reply:
I wasn't able to use `views_mut` because I later need the view id to later focus it in the editor
```rs
if let Some(id) = id {
    cx.editor.tree.focus = id;
}
```

Reply:
`view.id` is `Copy`, so you just need to dereference it so it gets copied and that way `let result` isn't bound to the tree.

Reply:
Will fix, thank you!